app/org/sagebionetworks/bridge/dynamodb/DynamoExternalIdDao.java

```diff
@@ -144,7 +146,8 @@ public final void setMapper(DynamoDBMapper mapper) {
 
         ForwardCursorPagedResourceList<ExternalIdentifierInfo> resourceList = new ForwardCursorPagedResourceList<>(
                 identifiers, offsetKey, pageSize)
-                .withFilter(ID_FILTER, idFilter);
+                .withFilter(ID_FILTER, idFilter)
+                .withFilter(TOTAL_FILTER,Integer.toString(identifiers.size()));
```

Comment:
Is there a unit test for this?

Also, I'm not sure this fixes the integ tests. It solves the NullPointerException, but the total will be the page size, while the integ test expects the count of all external IDs. In short, even with this fix, we still need to update the integ tests.